docs/source/query/overview.rst

```diff
@@ -5,6 +5,57 @@ Querying Snuba
 This guide drives you through the process of authoring and testing a
 Snuba query.
 
+
+Understanding the cost of queries
+=================================
+The cost of a snuba query has many factors but these are key to keep in mind when constructing queries:
+
+* Amount of rows scanned
+* Amount of columns scanned
+* Type of data scanned
+
+Let's take this query as an example::
+
+    MATCH events
+    SELECT count() AS count
+    WHERE 
+        timestamp > "2021-01-01" AND 
+        timestamp <= "2021-01-02" AND
+        project_id = 1
+ 
+The query above will scan two columns: timestamp and project id over one day. Timestamps and integers are also easy comparisons for a computer to make (can be done in one instruction). Thus this query is fairly cheap. The only way to make this query cheaper is to reduce the time window and scan fewer rows.
+
+Now let's see how we can make this query more expensive.
+
+Expanding the time frame::
+
+    MATCH events
+    SELECT count() AS count
+    WHERE 
+        timestamp > "2021-01-01" AND 
+        timestamp <= "2021-03-01" AND
+        project_id = 1
+ 
+
+Adding more fields to scan. Any extra fields that are scanned are extra files which have to be opened by clickhouse::
+
+    MATCH events
+    SELECT count() AS count
+    WHERE 
+        timestamp > "2021-01-01" AND 
+        timestamp <= "2021-03-01" AND
+        project_id = 1 AND
+        browser = "firefox" AND
+        os = "ios" AND
+        url = "https://santry.io/some_path" AND
+        tags[duration_group] = IN tuple("<10s", "<30s")
+        
+
+Nested fields are especially expensive to unnest and compare.
```

Comment:
You should clarify what a "nested" field is here.

Reply:
Nested fields are fairly complicated, I am not sure I would just brush them off as "especially expensive" as it is not exactly accurate,

The reason for this is that I saw a lot of scenarios with product teams accepting that something is "universally too expensive" and drawing the wrong conclusion that such feature is never to be used.

Specific cases:
- load tags for your results (even if you arrayJoin them) without filtering on them or aggregating a large amount of rows. This is not a problem as it happens after filtering. So you already restricted your dataset
- Aggregate over an arrayJoin of a nested column on a relatively short period of time. Not an issue. It largely depends if the dataset has enough shards.
- Search for the values of a specific tag (has(tags.key, 'asd')). This is generally good (at least unless every event has that field), That is because we have a bloomfilter index on the tags keys column
- Search on a specific tag key-value pair. Same there is a bloomfitler index. This is manageable over large period of times (unless all rows have it).
- regex on tags. This is deadly. we cannot use the index we need to unpack everything and then run a regex over every row.

So fundamentally are nested columns good or bad? it depends. I don't think we should take a universal position one way or another. Unfortunately for all practical purposes there is no simple one line rule that applies everything. In order to assess the impact of using them we need to know how they work and how they would work in the specific use case.